include/oneapi/dpl/pstl/utils.h

```diff
@@ -287,6 +287,42 @@ class __transform_functor
     }
 };
 
+template <typename __UnaryOper, typename __UnaryPred>
+class __transform_if_unary_functor
+{
+    mutable __UnaryOper _M_oper;
```

Comment:
Maybe there is a reason why but instead of making `_M_oper` and `_M_pred` mutable, can we just remove the const from the function call operator?

Reply:
Its a good question.  I'm following the lead of the rest of the functors in this file. 

Making the functor `const`, but the operator / predicate `mutable` would allow us to satisfy const requirements later on.  
I did some experimenting to confirm:
https://godbolt.org/z/848d1asT9

I checked using the test, and I don't believe there are `const` requirements elsewhere in the code that rely on this.  It would likely be safer to remove the const from the operator.

Reply:
Actually, let me correct my previous statement:  `walk_n` in `unseq_backend_sycl` does have const requirements which we use to implement the pattern, and beyond that `__parallel_for_submitter` and `parallel_for`.  For this reason, we need this functor to be `const`.
We also would prefer to not enforce a `const` requirement for the users' predicate and unary / binary operators.  To accomplish this, I believe it needs a `mutable` here for `_M_oper` and `_M_pred`.  Similar is done in this file for `__invoke_unary_op` and `__reorder_pred`.  

If we want to drop `mutable`, I think the only choice would be to impose the `const` requirement on the input operator and predicate.